src/niftyone/figures/generator.py

```diff
@@ -118,12 +119,15 @@ def __call__(
             records = [table.nested.loc[ind] for ind in inds]
             self.generate(records=records, out_dir=out_dir, overwrite=overwrite)
 
-    def _figure_name(self) -> None:
+    def _figure_name(self) -> dict[str, Any]:
```

Comment:
I think we should think of an alternative approach. To start, I think we should look for a way to avoid mutating `view_kwargs` altogether. Using it as mutable state, when it is intended as static pass-through kwargs is not great, and I think it leads to issues like we are facing here.

Reply:
I agree about the `view_kwargs` mutation - chipping away at this in #25. Should note that this fixes the mutation issue for the figure name, but I think will have a way soon to fix this issue with those changes as well.

I'm happy to punt this down to that PR, though I'd leave this open with the option to merge this in in the meantime if there is a need for a patch fix. The temporary workaround still requires the user to edit the installed code base.